### ISSUE TITLE
Allow pasting images in an encrypted note

### ISSUE THREAD

**Issue by thibaultmol** [meld-cp/obsidian-encrypt#170]
Is it possible to save an image (encrypted) inside of an encrypted note?

**Comment by MaelImhof** [comment]
It can be done manually, but the plugin could propose to do it automatically when right-clicking on the image for example.

Or even better, a setting to do that automatically when an image is added into an encrypted note.

**Comment by meld-cp** [comment]
An auto convert from clipboard to data:base64 image would be pretty neat. (It could even be a standalone plugin, if it doesn't already exist)

Source view would look quite bad if the user pastes in a large image though.

But I think it's worth an try 🤔

Drop in any use cases or implementation ideas here if you have them.

**Comment by thibaultmol** [comment]
@carlgustavh funnily enough, i was thinking of base64 as well but forget there was arleady a way to do it in obsidian.

but yeah as @MaelImhof and @meld-cp a slight automation would be nice. i'll do it manually for now

**Comment by meld-cp** [comment]
Yeah, I can imagine a workflow for pasting an image from the clipboard or dragging one in and have this (or another) plugin convert it into an embedded base64 image.


Curious if anyone is using the beta versions by chance?   I'd like to release it soon but I'm unsure how much it's been tested out in the wild.

**Comment by meld-cp** [comment]
I haven't tested it but found this after a quick browse through the plugins:

https://github.com/ZackaryW/obsidian-image-inline

It may work quite well with the new editor in the beta releases.

I might give it a try on the weekend.

**Comment by Kurandur** [comment]
@meld-cp The obsidian-image-inline plugin works perfectly for this. I only need this feature for encrypted notes. Would be interested in a pull-request, which integrates this into obsidian-encrypt? Then I would implement this.

**Comment by spookalucca** [comment]
I would love to see this be native in Meld Encrypt. As a new user, it feels weird to need another plugin just for images.
